contractsSol5/KyberStorage.sol

```diff
@@ -4,13 +4,109 @@ import "./IKyberStorage.sol";
 import "./IKyberDAO.sol";
 import "./IKyberFeeHandler.sol";
 import "./IKyberMatchingEngine.sol";
+import "./IKyberNetwork.sol";
+import "./utils/PermissionGroups2.sol";
 
-contract KyberStorage is IKyberStorage {
 
+/**
+ *   @title KyberStorage contract
+ *   Receives call from KyberNetwork for:
+ *   - record contract changes for matchingEngine, feeHandler, reserves and kyberDAO
+ */
+contract KyberStorage is IKyberStorage, PermissionGroups2 {
     // store current and previous contracts.
```

Comment:
not sure we need permission groups.

unless u set admin to be kyberNetwork
which is possible.
and then onlyAdmin can be used.
which is actually better.

can set admin in ctor.

Reply:
we want to have admin to set and update network, then network can set other address like dao, feeHandler,....

Reply:
change to constructor set the network address